### PR TITLE
WIP: First pass at animal classifier

### PR DESCRIPTION
**Pull request by kbrose**
First pass at animal classifier.

Achieves ~80% accuracy on the validation set after 10-20 epochs, but is pretty unstable.

WIP until the following are resolved:

* [ ] improve performance / stability
* [x] improve interface -- right now you just run the `classifier.py` file, but obviously that should be factored into the `main.py` file instead
* [ ] model save / load functionality
* [ ] standardize selection of classes -- I manually moved the classes with < 1000 images out of the folder (except the "empty" class), but this should probably be a parameter

This should be runnable on a CPU just fine, but it might be a little slow.

### CODE REVIEW

**Comment by lukeforehand** [comment]
Can't wait to take a look!  I wonder if my labeled "whitetail" set will help at all.  Labeled tons of bears and bucks, and lots of turkeys.

File: deertracker/classifier.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        return self.d2(x)
+
+
+def train(name: str, data_dir: Path = DEFAULT_DATA_FOLDER):
+    train_ds = tf.keras.preprocessing.image_dataset_from_directory(
+        data_dir,
+        validation_split=0.2,
+        subset="training",
+        seed=20201130,
+        image_size=(IMAGE_SIZE + 10, IMAGE_SIZE + 10),
```
Comment (lukeforehand) [inline deertracker/classifier.py:51]: Is this `+ 10` in adding padding, wondering because I padded the crops too, so making sure we aren't double padding if not intended.

File: deertracker/classifier.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        batch_size=1,
+    )
+    test_ds = tf.keras.preprocessing.image_dataset_from_directory(
```
Comment (lukeforehand) [inline deertracker/classifier.py:54]: Wow what a helpful function!  Code is working fine on my laptop testing the "whitetail" dataset.  Sweet :-)

**Comment by lukeforehand** [comment]
I added the cli entrypoint for the training, then I merged this to main.  I wasn't intending it to close this PR sorry about that.

**Comment by kbrose** [comment]
No worries!

File: deertracker/classifier.py (hunk 1 above)
Comment (kbrose) [reply deertracker/classifier.py:51]: Nope, this is giving head room for the random crop that happens later down in the `augment` function:
```python
image = tf.image.random_crop(image, [IMAGE_SIZE, IMAGE_SIZE, 3])
```
